nerfstudio/cameras/rays.py

```diff
@@ -138,6 +146,42 @@ def get_weights(self, densities: TensorType[..., "num_samples", 1]) -> TensorTyp
 
         return weights
 
+    @staticmethod
+    def get_weights_from_alphas(alphas: TensorType[..., "num_samples", 1]) -> TensorType[..., "num_samples", 1]:
+        """Return weights based on predicted alphas
+        Args:
+            alphas: Predicted alphas (maybe from sdf) for samples along ray
+        Returns:
+            Weights for each sample
+        """
+
+        transmittance = torch.cumprod(
+            torch.cat([torch.ones((*alphas.shape[:1], 1, 1), device=alphas.device), 1.0 - alphas + 1e-7], 1), 1
+        )
+
+        weights = alphas * transmittance[:, :-1, :]
```

Comment:
Can you call `get_weights_and_transmittance_from_alphas` but only return weights.